src/python/pants/backend/cc/target_types.py

```diff
@@ -51,11 +73,64 @@ class CCGeneratorFieldSet(FieldSet):
 # -----------------------------------------------------------------------------------------------
 
 
+class CCCompileFlagsField(StringSequenceField):
```

Comment:
@tdyas @Eric-Arellano @stuhood 
Big bikeshed on `alias` names - I've seen most permutations of flags/opts/options/some other things in the field. I like clear, concise, unambiguous... But that might almost be at a detriment because they look different to other tools.

Reply:
See https://github.com/pantsbuild/pants/discussions/16777